src/metabase/driver/h2.clj

```diff
@@ -121,29 +121,47 @@
       (when (instance? SessionLocal session)
         (Parser. session)))))
 
-(defn- contains-ddl?
-  [database query]
+(defn- check-disallow-ddl-commands
+  [{:keys [database] {:keys [query]} :native}]
   (when-let [h2-parser (make-h2-parser database)]
-    (try
-      (let [command      (.prepareCommand h2-parser query)
-            command-type (.getCommandType command)]
-        ;; TODO: do we need to handle CommandList?
-        ;; Command types are organized with all DDL commands listed first
-        ;; see https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/command/CommandInterface.java
-        (< command-type CommandInterface/ALTER_SEQUENCE))
-      ;; if the query is invalid, then it isn't DDL
-      (catch Throwable _ false))))
-
-(defn- check-disallow-ddl-commands [{:keys [database] {:keys [query]} :native}]
-  (when (and query (contains-ddl? database query))
-    (throw (IllegalArgumentException. "DDL commands are not allowed to be used with h2."))))
+    (when-let [command (try (.prepareCommand h2-parser query)
+                            ;; if the query is invalid, errors will get caught later
+                            (catch Throwable _ nil))]
+      (let [command-type (.getCommandType command)]
+        (cond
+          ;; if the command is a CommandList, then it is a multi-statement query
+          (= (type command) org.h2.command.CommandList)
```

Comment:
I assume you've tested this and figured it out, but I'm a little perplexed since `CommandList` only seems to exist in v. 1 ([javadoc link](https://javadoc.io/static/com.h2database/h2/1.2.126/org/h2/command/CommandList.html)) but according to deps.edn we're on version 2.1.212 which [does not have a `CommandList`](https://javadoc.io/static/com.h2database/h2/2.1.212/org/h2/command/package-summary.html)

Reply:
News to me. CommandList still exists, but it was changed from being a `public class` to just a `class` (not public). https://github.com/h2database/h2database/commit/16c099939efa010cf230187b37970eecef185137#diff-650e5c48e4aa26503577f72765d0c49f76eb197b205891048a19d1fc5f4a2fe8R16

Reply:
Doesn't feel great to use a package-private class, but I guess our hands are tied

Reply:
Maybe there's a more robust way to check whether there's multiple statements from the parser, but I couldn't find any. Before the upgrade to H2 v2.1 we were able to iterate over the commands with `(map class commands)`, but that's not possible anymore.